gui/wxpython/mapdisp/settings.py

```diff
@@ -0,0 +1,305 @@
+"""
+@package mapdisp.settings
+
+@brief Classes for map display settings management
+
+Classes:
+ - settings::ChBRender
+ - settings::ChBShowRegion
+ - settings::ChBAlignExtent
+ - settings::ChBResolution
+ - settings::MapDisplayPreferencesDialog
+
+(C) 2021 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author Vaclav Petras <wenzeslaus gmail.com>
+@author Anna Kratochvilova <kratochanna gmail.com>
+@author Linda Kladivova <lindakladivova gmail.com>
+"""
+
+import wx
+import wx.lib.scrolledpanel as SP
+from gui_core.wrap import StaticBox
+
+from core.settings import UserSettings
+from gui_core.preferences import PreferencesBaseDialog
+
+
+class ChBItem:
+    """Base class for checkbox settings items"""
+
+    def __init__(self, parent, mapframe):
+        self.mapFrame = mapframe
+        self._properties = self.mapFrame.mapWindowProperties
+        self.checkbox = wx.CheckBox(parent=parent, id=wx.ID_ANY, label=self.label)
+        self.checkbox.SetValue(self._property)
+        self.checkbox.SetToolTip(wx.ToolTip(self.tooltip))
+
+        self._connect()
+        self.checkbox.Bind(wx.EVT_CHECKBOX, self._onToggleCheckBox)
+
+    @property
+    def name(self):
+        pass
+
+    @property
+    def label(self):
+        pass
+
+    @property
+    def tooltip(self):
+        pass
+
+    @property
+    def _property(self):
+        pass
+
+    @_property.setter

```

Comment:
This probably shouldn't be private when you provide a setter. Or remove the setter.